Build Mono.AotCross error

### Description

Failed to build Mono.AotCross from the repository root directory

### Reproduction Steps

```bash
cd /path/to/runtime
./build.sh -subset Mono.AotCross 
```

### Expected behavior

Normal build project

### Actual behavior

```bash
➜ ./build.sh -subset Mono.AotCross                                            
  正在确定要还原的项目…
  Tool 'coverlet.console' (version '6.0.4') was restored. Available commands: coverlet
  Tool 'dotnet-reportgenerator-globaltool' (version '5.4.3') was restored. Available commands: reportgenerator
  Tool 'microsoft.dotnet.xharness.cli' (version '10.0.0-prerelease.25330.2') was restored. Available commands: xharness
  Tool 'microsoft.visualstudio.slngen.tool' (version '12.0.15') was restored. Available commands: slngen
  
  Restore was successful.
  所有项目均是最新的，无法还原。
  正在确定要还原的项目…
  所有项目均是最新的，无法还原。
  正在确定要还原的项目…
  所有项目均是最新的，无法还原。
/Users/lin/Files/Git/GitHub/runtime/src/mono/monoaotcross.proj(42,7): error MSB4184: 无法计算表达式“"".Substring(0, -1)”。length ('-1') must be a non-negative value. (Parameter 'length')
/Users/lin/Files/Git/GitHub/runtime/src/mono/monoaotcross.proj(42,7): error MSB4184: Actual value was -1.

生成失败。

/Users/lin/Files/Git/GitHub/runtime/src/mono/monoaotcross.proj(42,7): error MSB4184: 无法计算表达式“"".Substring(0, -1)”。length ('-1') must be a non-negative value. (Parameter 'length')
/Users/lin/Files/Git/GitHub/runtime/src/mono/monoaotcross.proj(42,7): error MSB4184: Actual value was -1.
    0 个警告
    1 个错误

已用时间 00:00:02.65
Build failed with exit code 1. Check errors above.
```

### Regression?

_No response_

### Known Workarounds

_No response_

### Configuration

SDK：10.0.100-preview.7.25322.101
macos 15.5 arm64

### Other information

_No response_

--- a/eng/Subsets.props
+++ b/eng/Subsets.props
@@ -558,7 +558,7 @@
   </ItemGroup>
 
   <ItemGroup Condition="$(_subset.Contains('+mono.aotcross+'))">
-    <ProjectToBuild Include="$(MonoProjectRoot)monoaotcross.proj" Category="mono" AdditionalProperties="%(AdditionalProperties);DotNetBuildMonoCrossAOT=true" />
+    <ProjectToBuild Include="$(MonoProjectRoot)monoaotcross.proj" Category="mono" />
   </ItemGroup>
 
   <ItemGroup Condition="$(_subset.Contains('+mono.corelib+'))">

--- a/eng/Subsets.props
+++ b/eng/Subsets.props
@@ -153,7 +153,7 @@
     <BootstrapSubsets Condition="'$(UseNativeAotForComponents)' == 'true'">$(BootstrapSubsets)+clr.nativeaotlibs+clr.nativeaotruntime+libs.native</BootstrapSubsets>
 
     <SwapNativeForIL Condition="$(_subset.Contains('+bootstrap+')) and '$(RuntimeFlavor)' == 'CoreCLR'">true</SwapNativeForIL>
-    
+
     <!-- Define AllSubsets to include all available subsets including OnDemand ones -->
     <AllSubsetsExpansion>clr.nativeprereqs+clr.iltools+clr.runtime+clr.native+clr.aot+clr.nativeaotlibs+clr.nativeaotruntime+clr.crossarchtools</AllSubsetsExpansion>
     <AllSubsetsExpansion>$(AllSubsetsExpansion)+clr.paltests+clr.paltestlist+clr.hosts+clr.jit+clr.alljits+clr.alljitscommunity+clr.spmi+clr.corelib+clr.nativecorelib+clr.tools+clr.toolstests+clr.packages</AllSubsetsExpansion>
@@ -558,7 +558,7 @@
   </ItemGroup>
 
   <ItemGroup Condition="$(_subset.Contains('+mono.aotcross+'))">
-    <ProjectToBuild Include="$(MonoProjectRoot)monoaotcross.proj" Category="mono" />
+    <ProjectToBuild Include="$(MonoProjectRoot)monoaotcross.proj" Category="mono" AdditionalProperties="%(AdditionalProperties);DotNetBuildMonoCrossAOT=true" />
   </ItemGroup>
 
   <ItemGroup Condition="$(_subset.Contains('+mono.corelib+'))">

--- a/src/mono/monoaotcross.proj
+++ b/src/mono/monoaotcross.proj
@@ -22,6 +22,15 @@
     <MonoAotTargetRids Include="$(MonoAotTargets.Split(';'))" />
   </ItemGroup>
 
+  <Target Name="CheckMonoAotTargets" BeforeTargets="BuildMonoCrossAllTargets">
+    <PropertyGroup>
+      <_SupportOS>android, browser, wasi</_SupportOS>
+      <_SupportOS Condition="$([MSBuild]::IsOSPlatform('OSX'))">$(_SupportOS), tvos, ios, maccatalyst</_SupportOS>
+    </PropertyGroup>
+
+    <Error Condition="'@(MonoAotTargetRids->Count())' == '0'" Text="MonoAotTargets is empty. Please configure MonoCrossAOTTargetOS with supported values: $(_SupportOS). Multiple values should be separated with '+' (e.g., /p:MonoCrossAOTTargetOS=android+browser)." />
+  </Target>
+
   <Target Name="BuildMonoCrossAllTargets" AfterTargets="Build">
     <MSBuild Targets="Restore"
              Projects="$(MSBuildThisFileDirectory)mono.proj"

